### PR TITLE
Deploy github site using maven-scm-publish plugin :building_construction: 

### PR DESCRIPTION
**Pull request by ivy-rew**
first approach that works without e-git; here to gather (your :wink: ) early feedback @alexsuter

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -27,18 +27,51 @@ pipeline {
...
+              sh """ 
+                git config --global user.name 'ivy-team'
+                git config --global user.email 'info@ivyteam.ch'
+
+                git clone -b gh-pages git@github.com:axonivy/project-build-plugin.git target/site-deployer
+                cd target/site-deployer
+                git checkout -b ${branch}
+                cd ../site
+                cp -r -v . ../site-deployer
+                cd ../site-deployer
```
Comment (ivy-rew) [inline Jenkinsfile:63]: nasty copying could be avoided by:

1. cloning the branch into target/site
2. building with maven: enriches this target/site dir
3. then again in git env: commit/push

would you prefer this?

File: Jenkinsfile [hunk 2]
```diff
@@ -27,18 +27,51 @@ pipeline {
...
+      steps {
+        script {
+          maven cmd: "clean site " +
+            "-Divy.engine.list.url=${params.engineListUrl} "
+
+          withEnv(['GIT_SSH_COMMAND=ssh -o StrictHostKeyChecking=no']) {
+            sshagent(credentials: ['github-axonivy']) {
+              def branch = "pages_$BRANCH_NAME-$BUILD_NUMBER"
+              def message = "Update site (${env.BRANCH_NAME})"
+              sh """
```
Comment (ivy-rew) [inline Jenkinsfile:54]: the bash script is quite large; can be shared in common .sh script;
as I'm going to use the same code also in the build/release/Jenkinsfile
the common script has the downside of not being editable on re-runs.
do you prefer extracting it anyway?

**Comment by alexsuter** [comment]
I think something like this looks easier:
https://www.lorenzobettini.it/2020/01/publishing-a-maven-site-to-github-pages/

**Review by alexsuter — COMMENTED** [review]
I think something like this looks easier:
https://www.lorenzobettini.it/2020/01/publishing-a-maven-site-to-github-pages/

**Comment by ivy-rew** [comment]
> I think something like this looks easier: https://www.lorenzobettini.it/2020/01/publishing-a-maven-site-to-github-pages/

definitively promising, thanks for sharing. I'll try this approach.

File: Jenkinsfile [hunk 3]
```diff
@@ -12,10 +12,6 @@ pipeline {
   }
 
   parameters {
-    booleanParam(name: 'skipGitHubSite',
-      description: 'If checked the plugin documentation on GitHub will NOT be updated',
-      defaultValue: true)
-
```
Comment (ivy-rew) [inline Jenkinsfile:18]: I think this one of the reasons why the release pipeline tends to fail: ... we did not automatically run github sites creation (due to its slow runtime with egit). 
Now we will update the snapshot docs on every nightly, which is what we want I think

File: Jenkinsfile [hunk 4]
```diff
@@ -27,18 +23,37 @@ pipeline {
...
+        script {
+          withEnv(['GIT_SSH_COMMAND=ssh -o StrictHostKeyChecking=no']) {
+            sshagent(credentials: ['github-axonivy']) {
+              def branch = "pages_$BRANCH_NAME-$BUILD_NUMBER"
+              def message = "Update site (${env.BRANCH_NAME})"
+              sh """ 
+                git config --global user.name 'ivy-team'
+                git config --global user.email 'info@ivyteam.ch'
+              """
+              maven cmd: "site site:stage scm-publish:publish-scm"
```
Comment (ivy-rew) [inline Jenkinsfile:51]: ci: also generates the site ... just before deploying

File: build/release/Jenkinsfile [hunk 5]
```diff
@@ -41,6 +41,8 @@ pipeline {
                   sh "gh auth login --with-token < ${tokenFile}"
                   sh "gh pr create --title '${message}' --body '${message}' --head ${targetBranch} --base ${env.BRANCH_NAME}"
                 }
+                // publish site from 'mvn release' run
+                maven cmd: "site:stage scm-publish:publish-scm"
```
Comment (ivy-rew) [inline build/release/Jenkinsfile:45]: no site generation (would be in wrong mode -> next snapshot).
I think the generated site from release-profile should still exist and be re-usable for publishing (but hard to test without actually running a release :)

File: pom.xml [hunk 6]
```diff
@@ -349,7 +333,7 @@
           <configuration>
             <tagNameFormat>v@{project.version}</tagNameFormat>
             <releaseProfiles>release</releaseProfiles>
-            <goals>deploy site-deploy</goals>
+            <goals>deploy site</goals>
```
Comment (ivy-rew) [inline pom.xml:336]: the actual xhtml generation when running a release.

**Comment by ivy-rew** [comment]
here's a sample site publish commit: provided by scm-publish:publish-scm (though run on my local cli)

https://github.com/axonivy/project-build-plugin/commit/f500991112c69ae66012801d6813021f7b455212

**Review by alexsuter — APPROVED** [review]
much better :+1:

**Pull request merged**